pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/util/DependencyOrder.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.gradle.python.util;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.artifacts.result.DependencyResult;
+import org.gradle.api.artifacts.result.ResolvedComponentResult;
+import org.gradle.api.artifacts.result.ResolvedDependencyResult;
+
+
+/**
+ * A container of static utilities for dependency order.
+ */
+public class DependencyOrder {
+    private DependencyOrder() {
+        // prevent instantiation of the utility class
+    }
+
+    /**
+     * Traverses the dependency tree post-order and collects dependencies.
+     * <p>
+     * The recursive post-order traversal returns the root of the (sub)tree.
+     * This allows the post-order adding into the set of dependencies as we
+     * return from the recursive calls. The set of seen dependencies ensures
+     * ensures that the cycles in Ivy metadata do not cause cycles in our
+     * recursive calls.
+     *
+     * @param root the root of the dependency (sub)tree
+     * @param seen the input set of already seen dependencies
+     * @param dependencies the output set of dependencies in post-order
+     * @return the root itself
+     */
+    public static ResolvedComponentResult postOrderDependencies(
+        ResolvedComponentResult root,
+        Set<ComponentIdentifier> seen,
+        Set<ResolvedComponentResult> dependencies
+    ) {
```

Comment:
They don't fit in 120 characters margin.
That's why I put them on separate lines.